### DISCUSSION TITLE
Add the ability to Log Laravel Events

### DISCUSSION THREAD

**Discussion by lukebouch** [spatie/laravel-activitylog#1109] [Ideas]
I'm working on an app that I need to log a bunch of user performed actions and it made me wonder, should I create events for every action and then add a listener to log it. This made me think, it would be great to add the ability to log events directly to the package. Would that be a useful feature?

I could see there being a contract or trait you could add to an event to be able to define how and when it should be logged. I'm open to feedback as this maybe be a small use-case.

**Comment by Gummibeer** [comment]
I think it's a pretty common one. But also super subjective and the way they are logged and what changes dramatically from project to project.

In that case we would also get pretty fast in big trouble regarding event sourcing.
Since we've added and improved model event logging we receive requests to add a rollback functionality.
Would be the same for events. The moment we add a trait or listener to log events we would receive requests to allow to replay them and so on. And that's what eventsourcing is about.

So Feature wise - this package is pretty feature complete. So not saying we can freeze it. But it's more about improving the existing features than adding real new ones. The batch feature was one of the big improvements and somehow new features.
